src/datatypes/src/vectors/constant.rs

```diff
@@ -83,6 +84,34 @@ impl ConstantVector {
             self.length,
         )))
     }
+
+    pub(crate) fn take_vector(&self, indices: &UInt32Vector) -> Result<VectorRef> {
+        if indices.is_empty() {
+            return Ok(self.slice(0, 0));
+        }
```

Comment:
correctness: If the `indices` slot is null, arrow's take returns a null value. So the length of the result is equal to `indices.len()`. However, we could return error for a constant vector when indices contain null since constant vector doesn't support null.
```
ensure!(indices.null_count() == 0, error);
```

Reply:
Nice catch, thanks